Add KuzzleRequest.getSearchBody documentation (#2046)
kuzzleio/kuzzle: doc/2/framework/classes/kuzzle-request/get-search-body/index.md

---
code: true
type: page
title: getSearchBody
description: KuzzleRequest class getSearchBody() method
---

# getSearchBody

<SinceBadge version="2.11.0" />

Returns the search body.

> With the HTTP protocol, the search body will be fetch depending on the verb:

Reviewer: ```suggestion
In the HTTP protocol, the search body will be extracted depending on the verb:
```

  - `GET`: extract the `searchBody` query string argument and parse it
  - `POST`: return the request body

> Typically the search body contains keywords used by Elasticsearch to perform queries like `query`, `sort`, `aggregations` etc.

Reviewer: I don't understand the purpose of this sentence.

Author: People are often confused about what is inside the search body (one of our client for example)

> Typically the search body contains keywords used by Elasticsearch to perform queries like `query`, `sort`, `aggregations` etc.

Author: ```suggestion
Typically the search body contains keywords used by Kuzzle to perform queries like `query`, `sort`, `aggregations` etc.
```
